system/include/emscripten/val.h

```diff
@@ -189,10 +190,15 @@ namespace emscripten {
             union {
                 unsigned u;
                 float f;
+                #if __ILP32__
                 const void* p;
+                #endif
             } w[2];
             double d;
             uint64_t u;
+            #if __LP64__
+            const void* p2;
```

Comment:
It looks like the author of this code was ok with field names being duplicated between inner and outer union (e.g. `u` is used in both) so maybe we should continue that pattern?